File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        lo, hi = 0, len(height) - 1
        result = 0

        while lo < hi:
            result = max(result, (hi - lo) * min(height[lo], height[hi]))

            if height[lo] < height[hi]:
                lo += 1
            else:
                hi -= 1

        return result

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int lo = 0;
        int hi = height.length - 1;
        int water = 0;

        while (lo < hi) {
            water = Math.max(water, (hi - lo) * Math.min(height[lo], height[hi]));

            if (height[lo] < height[hi]) {
                lo++;
            } else {
                hi--;
            }
        }
        
        return water;
    }
}